uaclient/contract_data_types.py

```diff
@@ -0,0 +1,572 @@
+from typing import List, Optional
+
+from uaclient.data_types import (
+    BoolDataValue,
+    DataObject,
+    Field,
+    IntDataValue,
+    StringDataValue,
+    data_list,
+)
+
+
+class ActivityInfo(DataObject):
+    fields = [
+        Field("activityID", StringDataValue, False),
+        Field("activityToken", StringDataValue, False),
+        Field("activityPingInterval", IntDataValue, False),
+        Field("resources", data_list(StringDataValue), False),
+    ]
+
+    def __init__(
+        self,
+        activityID: Optional[str],
+        activityToken: Optional[str],
+        activityPingInterval: Optional[int],
+        resources: Optional[List[str]],
+    ):
+        self.activityID = activityID
+        self.activityPingInterval = activityPingInterval
+        self.activityToken = activityToken
+        self.resources = resources
+
+
+class AvailableResource(DataObject):
+    fields = [
+        Field("available", BoolDataValue, False),
+        Field("name", StringDataValue, False),
+        Field("description", StringDataValue, False),
+        Field("presentedAs", StringDataValue, False),
+    ]
+
+    def __init__(
+        self,
+        available: Optional[bool],
+        name: Optional[str],
+        description: Optional[str],
+        presentedAs: Optional[str],
+    ):
+        self.available = available
+        self.name = name
+        self.description = description
+        self.presentedAs = presentedAs
+
+
+class ExternalID(DataObject):
+    fields = [
+        Field("origin", StringDataValue, False),
+        Field("IDs", data_list(StringDataValue), False),
+    ]
+
+    def __init__(self, origin: Optional[str], IDs: Optional[List[str]]):
+        self.origin = origin
+        self.IDs = IDs
+
+
+class AccountInfo(DataObject):
+    fields = [
+        Field("name", StringDataValue, False),
+        Field("id", StringDataValue, False),
+        Field("createdAt", StringDataValue, False),
+        Field("type", StringDataValue, False),
+        Field("userRoleOnAccount", StringDataValue, False),
+        Field("externalAccountIDs", data_list(ExternalID), False),
+    ]
+
+    def __init__(
+        self,
+        name: Optional[str],
+        id: Optional[str],
+        createdAt: Optional[str],
+        type: Optional[str],
+        userRoleOnAccount: Optional[str],
+        externalAccountIDs: Optional[ExternalID],
+    ):
+        self.name = name
+        self.id = id
+        self.createAt = createdAt
+        self.type = type
+        self.userRoleOnAccount = userRoleOnAccount
+        self.externalAccountIDs = externalAccountIDs
+
+
+class ActiveMachine(DataObject):
```

Comment:
A user may not want this information to be non-root readable either. And we don't use it. Let's not include it

Reply:
Okay. Makes sense.